### PR TITLE
feat: Alert before deleting or leaving the audio room

### LINKED ISSUE #160: Audio room is deleted without confirmation

### PR DESCRIPTION
**Pull request by AKASHANGADII**
closes #160

### CODE REVIEW

**Comment by chandansgowda** [comment]
@AKASHANGADII Please add a screen recording of the same.

**Comment by chandansgowda** [comment]
@AKASHANGADII The flow seems to be wrong. Is the room getting deleted properly ??
Coz we are going back and then trying to delete it .

**Comment by AKASHANGADII** [comment]
Yeah @chandansgowda I have maintained the same code flow.

**Comment by chandansgowda** [comment]
Yes. But we are navigating to the previous screen before deleting the room. Might lead to some inconsistencies.

**Comment by AKASHANGADII** [comment]
But I have written code to navigate only after deleting the controller.

**Comment by chandansgowda** [comment]
@AKASHANGADII If I'm not wrong, I'm not able to see a circular indicator in the single room screen once the delete dialog is confirmed.

**Comment by AKASHANGADII** [comment]
Okay I'll check it.

**Comment by AKASHANGADII** [comment]
Hey @chandansgowda .I have added the loading indicator. Please review it.

File: lib/controllers/single_room_controller.dart [hunk 1]
```diff
@@ -177,13 +179,27 @@ class SingleRoomController extends GetxController {
   }
 
   Future<void> leaveRoom() async {
+    Get.dialog(
```
Comment (chandansgowda) [inline lib/controllers/single_room_controller.dart:182]: Please refactor it as a widget. We might possibly use it many times.

**Review by chandansgowda — CHANGES_REQUESTED** [review]

**Comment by AKASHANGADII** [comment]
@chandansgowda . I have made the changes. Please look into it.

**Review by chandansgowda — APPROVED** [review]

**Comment by chandansgowda** [comment]
Great job @AKASHANGADII 🚀

**Pull request merged**